src/cr/cube/crunch_cube.py

```diff
@@ -149,7 +149,7 @@ def as_array(self, include_missing=False, weighted=True, adjusted=False,
     @lazyproperty
     def ca_dim_ind(self):
         for (i, dim) in enumerate(self.dimensions):
-            if dim.type == 'categorical_array':
+            if dim.dimension_type == 'categorical_array':
```

Comment:
I renamed `Dimension.type` to `.dimension_type`. There are some comments in the commit going into the rationale. Short version is having a property or method named `.type` or `.id` is to be avoided. Among other benefits this lets the reader know what kind of type it is when context can't make that clear.